redaxo/src/core/lib/base/factory_trait.php

```diff
@@ -55,15 +55,25 @@ public static function setFactoryClass($subclass)
     /**
      * Returns the class for the factory.
      *
-     * @return string
-     * @psalm-return class-string<static>
+     * @return class-string<static>
      */
     public static function getFactoryClass()
     {
         $calledClass = static::class;
         return self::$factoryClasses[$calledClass] ?? $calledClass;
     }
 
+    /**
+     * Returns the explicitly set factory class, otherwise null.
+     *
+     * @return class-string<static>|null
+     */
+    public static function getFactoryClassOrNull(): ?string
```

Comment:
Gibt es Alternativvorschläge für den Namen?
Also `getFactoryClass` liefert ja immer eine Klasse, im Zweifel die Klasse selbst, wenn keine Subklasse als FactoryClass gesetzt.
Und die neue Methode hier liefert halt nur was, wenn explizit eine andere Subklasse als factoryClass gesetzt wurde.
Daher wäre vielleicht auch `getChildFactoryClass`  denkbar.

Reply:
vllt die funktionalität via optionalem `$default` param in `getFactoryClass()` implementieren, statt separater methode?

Reply:
Hmm, so einen allgemeinen `$default`-Parameter finde ich dort tendenziell unpassend.
Schwierig wäre auch der Defaultwert für den `$default`-Param. Denn default wäre ja "called class" (bisheriger Default), und man könnte nicht `null` als Repräsentanz dafür nehmen, denn `null` wäre ja gerade der alternative Default-Wert.

Wenn man es in die vorhandene Methode integrieren möchte, dann glaube ich besser über einen boolschen zweiten Param, der die zwei Varianten steuert.
Wobei wir normalerweise ja separate Methoden bevorzugen vor boolschen Params.

Weiterer Namensvorschlag wäre noch `getExplicitFactoryClass`.

Reply:
> Weiterer Namensvorschlag wäre noch getExplicitFactoryClass.

👍